tests/kernel/threads/custom_data/custom_data_api/src/test_customdata_api.c

```diff
@@ -58,9 +58,20 @@ void test_customdata_get_set_preempt(void)
 	/** TESTPOINT: custom data of preempt thread */
 	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
 		customdata_entry, NULL, NULL, NULL,
-		K_PRIO_PREEMPT(0), 0, 0);
+		K_PRIO_PREEMPT(0), K_USER, 0);
 	k_sleep(500);
 
 	/* cleanup environment */
 	k_thread_abort(tid);
 }
+
+/*test case main entry*/
+void test_main(void)
+{
+	k_thread_access_grant(k_current_get(), &tdata, tstack, NULL);
+
+	ztest_test_suite(test_customdata_api,
+		ztest_unit_test(test_customdata_get_set_coop),
```

Comment:
any reason for not running the COOP test in user mode?

Reply:
user threads can't create new threads that are of greater priority than they are.